app/src/main/java/com/framgia/forder/utils/searchview/DataHelper.java

```diff
@@ -0,0 +1,105 @@
+package com.framgia.forder.utils.searchview;
+
+/**
+ * Copyright (C) 2015 Ari C.
+ * <p/>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import android.content.Context;
+import android.widget.Filter;
+import com.framgia.forder.data.model.DataSuggest;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class DataHelper {
+    private List<DataSuggest> mProducts = new ArrayList<>(
+            Arrays.asList(new DataSuggest("Com tấm"), new DataSuggest("Nước Chanh"),
+                    new DataSuggest("Sinh Tố"), new DataSuggest("Mì Quảng"),
+                    new DataSuggest("sữa coffe"), new DataSuggest("Trà sữa alisan"),
+                    new DataSuggest("Trà sữa Gong Cha"), new DataSuggest("Mì Quảng Ếch"),
+                    new DataSuggest("Chè Đậu Thập Cẩm"), new DataSuggest("Dừa Bến Tre"),
+                    new DataSuggest("Rau Câu Flan Trái Dừa"), new DataSuggest("Nước sấu-15"),
+                    new DataSuggest("Mì Que"), new DataSuggest("Trà sữa BoBaPop"),
+                    new DataSuggest("Trà sữa trân châu đen"), new DataSuggest("Cà Rốt Ép"),
+                    new DataSuggest("Chè Đậu Xanh Đánh"), new DataSuggest("Dâu Xay Sữa Chua")));
+
+    public void setDataSuggest(List<DataSuggest> products) {
+        mProducts = products;
+    }
+
+    public void resetSuggestionsHistory() {
+        for (DataSuggest dataSuggest : mProducts) {
+            dataSuggest.setIsHistory(false);
+        }
+    }
+
+    public void findSuggestions(Context context, String query, final int limit,
+            final long simulatedDelay, final OnFindSuggestionsListener listener) {
+        new Filter() {
+
+            @Override
+            protected FilterResults performFiltering(CharSequence constraint) {
+
+                try {
+                    Thread.sleep(simulatedDelay);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+
+                resetSuggestionsHistory();
+                List<DataSuggest> suggestionList = new ArrayList<>();
+                if (!(constraint == null || constraint.length() == 0)) {
+
+                    for (DataSuggest suggestion : mProducts) {
+                        if (suggestion.getBody().toUpperCase()
+                                .startsWith(constraint.toString().toUpperCase())) {
```

Comment:
WARNING: [DefaultLocale] Implicitly using the default locale is a common source of bugs: Use `toUpperCase(Locale)` instead. For strings meant to be internal use `Locale.ROOT`, otherwise `Locale.getDefault()`.
 Calling `String#toLowerCase()` or `#toUpperCase()` *without specifying an explicit locale* is a common source of bugs. The reason for that is that those methods will use the current locale on the user's device, and even though the code appears to work correctly when you are developing the app, it will fail in some locales. For example, in the Turkish locale, the uppercase replacement for `i` is *not* `I`.

If you want the methods to just perform ASCII replacement, for example to convert an enum name, call `String#toUpperCase(Locale.US)` instead. If you really want to use the current locale, call `String#toUpperCase(Locale.getDefault())` instead.